### PR TITLE
Systematize tests with url processing with the flag "i" (iframe) (close #869)

### LINKED ISSUE #869: Systematize tests with url processing with the flag "i" (iframe)
Cases of using the `i` flag:
- Assign src to iframe
- Assign a url attribute to elements (`a`, `form`, `area`, `base`) with the `target` attribute (`_blank`, `_self`, `_parent`, `_top` or framename)
- Assign the `target` attribute to elements with a url attribute
- Change iframe name
  - some `target` attributes cease to point to this iframe 
  - some `target` attributes start to point to this iframe
- Change the `target` attribute in the `base` tag
- Change a url via `location` (`href`, `path`, `perlace()`, `assign()`)
  - from top window into iframe
  - from iframe into iframe
  - from top window into cross-domain iframe
  - from iframe into cross-domain iframe

### PR DESCRIPTION
**Pull request by LavrovArtem**
/cc @churkin @miherlosev

### CODE REVIEW

**Comment by LavrovArtem** [comment]
FPR

File: src/processing/dom/base-dom-adapter.js [hunk 1]
```diff
@@ -3,7 +3,7 @@
 // Do not use any browser or node-specific API!
 // -------------------------------------------------------------
 
-const IFRAME_FLAG_TAGS = ['a', 'form'];
+const IFRAME_FLAG_TAGS = ['a', 'form', 'area'];
```
Comment (miherlosev) [inline src/processing/dom/base-dom-adapter.js]: I believe we can use https://github.com/DevExpress/testcafe-hammerhead/blob/master/src/processing/dom/index.js#L37

File: test/client/fixtures/sandbox/iframe-flag-test.js [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+});
+
+QUnit.testDone(function () {
+    iframeSandbox.off(iframeSandbox.RUN_TASK_SCRIPT, initIframeTestHandler);
+});
+
+function hasIframeFlag (url) {
+    var resourceType = urlUtils.parseProxyUrl(url).resourceType;
+
+    return !!resourceType && resourceType.indexOf('i') !== -1;
```
Comment (miherlosev) [inline test/client/fixtures/sandbox/iframe-flag-test.js]: Why do we need double negative - `!!resourceType`? 
Use `return resourceType && resourceType.indexOf('i') !== -1`

File: test/client/fixtures/sandbox/iframe-flag-test.js [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+    base: 'href'
+};
+
+module('target');
+
+test('assign a url attribute to elements with the "target" attribute in top window', function () {
+    var url    = 'https://example.com/';
+    var iframe = document.createElement('iframe');
+
+    /* eslint-disable camelcase */
```
Comment (miherlosev) [inline test/client/fixtures/sandbox/iframe-flag-test.js]: We can move eslint rule to the `.eslintrc` for tests (https://github.com/DevExpress/testcafe-hammerhead/blob/master/test/client/.eslintrc)

File: test/client/fixtures/sandbox/iframe-flag-test.js [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+var tagUrlAttr = {
+    a:    'href',
+    form: 'action',
+    area: 'href',
+    base: 'href'
+};
+
+module('target');
+
+test('assign a url attribute to elements with the "target" attribute in top window', function () {
```
Comment (miherlosev) [inline test/client/fixtures/sandbox/iframe-flag-test.js]: Move all `target` specific tests to the https://github.com/DevExpress/testcafe-hammerhead/blob/master/test/client/fixtures/target-attr-test.js

File: src/processing/dom/base-dom-adapter.js (hunk 1 above)
Comment (LavrovArtem) [inline src/processing/dom/base-dom-adapter.js]: No, I don't think so because we don't need to append the iframe flag to base element.

Comment (LavrovArtem) [inline src/processing/dom/base-dom-adapter.js]: Hmm or you offer ```IFRAME_FLAG_TAGS = TARGET_ATTR_TAGS.map(tagName => tagName !== 'base')```

File: test/client/fixtures/sandbox/iframe-flag-test.js (hunk 2 above)
Comment (LavrovArtem) [inline test/client/fixtures/sandbox/iframe-flag-test.js]: ok

File: test/client/fixtures/sandbox/iframe-flag-test.js (hunk 4 above)
Comment (LavrovArtem) [inline test/client/fixtures/sandbox/iframe-flag-test.js]: But I tests the iframe flag

File: test/client/fixtures/sandbox/iframe-flag-test.js (hunk 3 above)
Comment (LavrovArtem) [inline test/client/fixtures/sandbox/iframe-flag-test.js]: I think it's unnecessary

Comment (churkin) [inline test/client/fixtures/sandbox/iframe-flag-test.js]: @miherlosev Why? it is necessary only in few places

**Comment by LavrovArtem** [comment]
FPR

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
FPR

**Pull request merged**